src/pcm/SoxrResampler.cxx

```diff
@@ -109,14 +207,19 @@ SoxrPcmResampler::Open(AudioFormat &af, unsigned new_sample_rate)
 	assert(audio_valid_sample_rate(new_sample_rate));
 
 	soxr_error_t e;
+	soxr_io_spec_t* p_soxr_io = nullptr;
+	if(soxr_io.scale != 0) {
```

Comment:
This is a bad check for "using custom settings". This happens to work in practice, but depends on implicit zero initialization of a floating point in a global variable - which is obscure. Better be explicit here about checking `SOXR_CUSTOM_RECIPE`.

Reply:
I will make sure that the  struct is initialized, else an additional variable would be required.